pkg/process/checks/process.go

```diff
@@ -147,6 +150,11 @@ func (p *ProcessCheck) Init(syscfg *SysProbeConfig, info *HostInfo, oneShot bool
 
 	p.initConnRates()
 
+	serviceExtractorEnabled := true
```

Comment:
why is it enabled by default (and hard coded)?

Reply:
Enabling the feature by default and having "process_context" tag for processes can affect USM's service naming.

Reply:
We are modifying the agent-payload for process. We are including `processContext` into the payload. The logic for extracting `processContext` is in the serviceExtractor, which is used in other checks like the connections checks. For the purposes of the process check, we want to always enable the serviceExtractor because we always want to populate the `processContext` data.

Reply:
wouldn't we want to enable this only if `process_service_inference` is enabled? 
As far as I know the current behavior only adds `process_context` tags if that feature is explicitly enabled in the configs. This means that if this feature isn't enabled we won't be getting `process_context` and we don't need to populate that field (we don't need serviceExtractorEnabled to be true) since we won't be gathering that data.

Reply:
As discussed in the slack thread, me and Guy discussed and decided is safe to default true for the agent changes. Will be cautious when integrating this change to the backend.